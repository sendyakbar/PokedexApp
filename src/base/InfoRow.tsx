import React from 'react';
import {StyleSheet, View, Text, ViewStyle, TextStyle} from 'react-native';

type Props = {
  label: string;
  value: string;
};

export default function InfoRow(props: Props): JSX.Element {
  const {label, value} = props;

  return (
    <View style={styles.container}>
      <View style={styles.labelCol}>
        <Text style={styles.labelText}>{label}</Text>
        <Text style={styles.labelText}>:</Text>
      </View>
      <View style={styles.valueCol}>
        <Text style={styles.valueText}>{value}</Text>
      </View>
    </View>
  );
}

type StyleType = {
  container: ViewStyle;
  labelCol: ViewStyle;
  valueCol: ViewStyle;
  labelText: TextStyle;
  valueText: TextStyle;
};

const styles = StyleSheet.create<StyleType>({
  container: {
    flexDirection: 'row',
    paddingVertical: 2,
  },
  labelCol: {
    flex: 1,
    paddingLeft: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  valueCol: {
    flex: 2,
    paddingHorizontal: 8,
  },
  labelText: {
    fontSize: 14,
    color: 'black',
    fontWeight: '700',
  },
  valueText: {
    fontSize: 14,
    color: 'grey',
    fontWeight: '500',
  },
});
