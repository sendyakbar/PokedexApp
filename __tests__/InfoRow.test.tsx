import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import InfoRow from '../src/base/InfoRow';

it('renders correctly', () => {
  const tree = renderer
    .create(<InfoRow label="Label" value="Value" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
