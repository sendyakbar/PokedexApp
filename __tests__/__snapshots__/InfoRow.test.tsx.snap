// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "paddingVertical": 2,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "700",
        }
      }
    >
      Label
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
          "fontWeight": "700",
        }
      }
    >
      :
    </Text>
  </View>
  <View
    style={
      {
        "flex": 2,
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "grey",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Value
    </Text>
  </View>
</View>
`;
